src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter_Statics.cs

```diff
@@ -51,8 +51,9 @@ public static void CompileMethod(WebAssemblyCodegenCompilation compilation, WebA
             try
             {
                 string mangledName;
-                // TODO: We should use the startup node to generate StartupCodeMain and avoid special casing here
-                if (methodCodeNodeNeedingCode.Method.Signature.IsStatic && methodCodeNodeNeedingCode.Method.Name == "Main")
+
+                // TODO: Better detection of the StartupCodeMain method
+                if (methodCodeNodeNeedingCode.Method.Signature.IsStatic && methodCodeNodeNeedingCode.Method.Name == "StartupCodeMain")
```

Comment:
In debug builds, this causes tests to write to null on exit because of #5173. I'll put in a workaround if that isn't fixed by the time this change is ready to merge.

Reply:
Fixing #5173 in this change